src/components/Board.js

```diff
@@ -4,21 +4,26 @@ import Square from './Square';
 import PropTypes from 'prop-types';
 
 
-const generateSquareComponents = (squares, onClickCallback) => {
-  // Complete this for Wave 1
-  // squares is a 2D Array, but 
-  //  you need to return a 1D array
-  //  of square components
+const generateSquareComponents = (squares, onClickCb) => {
+  let sa = [].concat(...squares);
 
+  let mapped = sa.map((square) =>{
+    const clickHandler = square => {
+      onClickCb(square, sa);
+    };
+    return(
+    <Square key={square.id} id={square.id} onClickCallback={clickHandler} value={square.value} />
```

Comment:
one thing we can do to make this more readable is:

```suggestion
    <Square 
        key={square.id} 
        id={square.id} 
        onClickCallback={clickHandler} 
        value={square.value} 
    />
```